wormhole-connect/src/store/transferInput.ts

```diff
@@ -182,28 +182,62 @@ const performModificationsIfToChainChanged = (state: TransferInputState) => {
   }
 };
 
+// This function was relocated from a useEffect in the Bridge component to avoid
+// the visual delay of two store updates.
+// At the time, the dependency array included:
+// [showValidationState, availableRoutes, fromChain, toChain, token, destToken, amount]
```

Comment:
Good catch. I'd guess the reason this deps array included these things beyond `availableRoutes` must be to trigger the route selection whenever they are changed, but these same dependencies are already on the hook that sets `availableRoutes`, so it was kind of redundant.